crates/ruff_python_ast/src/comparable.rs

```diff
@@ -1,25 +1,19 @@
 //! An equivalent object hierarchy to the `RustPython` AST hierarchy, but with the
 //! ability to compare expressions for equality (via [`Eq`] and [`Hash`]).
+//!
+//! Two [`ComparableExpr`]s are considered equal if the underlying AST nodes have the
+//! same shape, ignoring trivia (e.g., parentheses, comments, and whitespace), the
+//! location in the source code, and other contextual information (e.g., whether they
```

Comment:
Thanks. Can we be a bit more explicit about the same shape. Let's say we change our string types to store the implicit concatenation parts. Would two strings with different parts then be considered unequal because the AST now has a different shape?